enhancements/support-less-on-more.md

```diff
@@ -0,0 +1,146 @@
+---
+status: propose
+title: Update OpenShift Pipelines support policy to support less versions on more OpenShift versions
+creation-date: '2023-07-31'
+last-updated: '2023-07-31'
+authors: ['vdemeester']
+---
+
+# Update OpenShift Pipelines support policy to support less versions on more OpenShift versions
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+  - [Support only the last 2 releases](#support-only-the-last-2-releases)
+  - [Support down to <code>MIN_KUBERNETES_VERSION</code>](#support-down-to-)
+- [Testing](#testing)
+- [Dependencies](#dependencies)
+- [Examples](#examples)
+- [When ?](#when-)
+<!-- /toc -->
+
+---
+type: project
+deadline: 2023-09-02
+tags:
+  - project
+---
+## Summary
+
+The plan is to reduce the number of OpenShift Pipelines version we
+support at a given time while augmenting the number of OpenShift
+version we support them in.
+
+
+Supporting Materials:
+ * [Enhance Support Policy of OpenShift Pipelines](./enhance-support-policy.md)
+ * [OCP 4.y lifecycles](https://docs.google.com/document/d/1exNNJPdwCH3F5eEKaLVBcrj_qAHfJ3PlfnquP0-6TxM/edit#heading=h.539ohzh6t1k6)
+ * [Proposal for Pipelines &
+   GitOps](https://docs.google.com/spreadsheets/d/1Tn6BHNBJnEaTBRm22U0KrNCR_Zwu_qBn-IR-xFEa4Os/edit#gid=1430910332)
+
+## Motivation
+
+OpenShift Pipelines versions are currently only shipped and supported
+for a small set of OpenShift version (n-1, n, n+1).  OpenShift support
+can be very long (especially for EUS version), which makes us having
+to support a lot of version at the same time. Adding to this support
+exception that we may give to customers, it can become a burden for
+maintenance, and a lot of bug-fix version to release.
+
+For example, today (2023, 31st of July), we are *technically*
+supporting the following:
+- 1.8.x on 4.10 (EUS)
+- 1.9.x on >= 4.10
+- 1.10.x
+- 1.11.x
+
+In addition, it has no real technological reason for not supporting
+multiple version. Tekton supports a relatively wide range of Kubernetes
+version, and so can OpenShift Pipelines.
+
+Enabling customers to get the latest of feature from Tekton in their
+supported cluster is a win-win situation : customers can use the
+latest, and we can ship the latest (fixes, …) to our customers.
+
+## Proposal
+
+### Support only the last 2 releases
+
+The proposal is to only support the last 2 releases. As of today, this
+would mean supporting 1.11 and 1.10.
+
+In addition, as soon as release N is released, release N-1 becomes in
+maintenance mode, aka gets only critical bugfixes and security
```

Comment:
Koustav brought up that we expect customers to deploy N and N-1 in prod and it's not the best scenario to not support N-1 in prod and pushing the upgrade to N all the times; we're already reducing release support footprint to 2 releases, but we need to be decently active on the N-1 release as well

Reply:
Define "decently active". We can also define what maintenance support means for us btw. We won't push new features for N-1 release, but we will backport bugfix is affected, in theory, we would do bugfix release when upstream LTS also gets bugfix releases, etc…

Reply:
I personally don't care how we call it officially. What I care is the amount of work we actually do - IMHO we should limit it as much as possible for N-1